src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -304,11 +308,10 @@ void Reorder::createReorderPrimitive(const dnnl::memory::desc& srcDesc,
         src_desc = src_blocked->getPrimitive().get_desc();
     }
 
+    DEBUG_LOG("CreateReorderPrimitive is called for node", getName(), " src desc: ", src_desc, " dst_desc: ", dst_desc);
+    CPU_NODE_ASSERT(src_desc.get_ndims() == dst_desc.get_ndims(), "OneDNN doesn't support reorder with different ranks.");
```

Comment:
Reorder node doesn't support input/output tensors with different ranks

Reply:
Actually there is at least one [exception](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/reorder.cpp#L290-L293), in which Reorder cpu node with different ranks might exist. This seems like a hack, but since it exists, I decided to add this check in order to make sure that `src` and `dst` descs have equal ranks after this hack.

Reply:
I actually proposed to update the comment only. But I agree: it wasn't obvious :)